plugin/src/py/android_screenshot_tests/test_recorder.py

```diff
@@ -243,6 +243,221 @@ def test_verify_failure(self):
             self.assertEqual((0, 128, 0, 255), im.getpixel((1, 1)))
             self.assertEqual((0, 128, 0, 255), im.getpixel((9, 1)))
 
+    def test_verify_different_sizes_longer(self):
+        self.create_temp_image("foobar.png", (10, 10), "blue")
+        self.make_metadata(
+            # language=json
+            """
+            [
+                {
+                    "name": "foobar",
+                    "tileWidth": 1,
+                    "tileHeight": 1
+                }
+            ]"""
+        )
+
+        self.recorder.record()
+        os.unlink(join(self.inputdir, "foobar.png"))
+        self.create_temp_image("foobar.png", (10, 20), "blue")
+
+        try:
+            self.recorder.verify()
+            self.fail("expected exception")
+        except VerifyError:
+            pass  # expected
+
+        self.assertTrue(os.path.exists(join(self.failureDir, "foobar_actual.png")))
+        self.assertTrue(os.path.exists(join(self.failureDir, "foobar_expected.png")))
+        self.assertTrue(os.path.exists(join(self.failureDir, "foobar_diff.png")))
+
+        # check colored diff
+        with Image.open(join(self.failureDir, "foobar_diff.png")) as im:
+            (w, h) = im.size
+            self.assertEqual(10, w)
+            self.assertEqual(20, h)
+
+            self.assertEqual((0, 0, 255, 255), im.getpixel((0, 0)))
+            self.assertEqual((0, 0, 255, 255), im.getpixel((9, 0)))
+            self.assertEqual((0, 0, 255, 255), im.getpixel((5, 0)))
+            self.assertEqual((0, 0, 255, 255), im.getpixel((0, 5)))
+
+            self.assertEqual((0, 0, 255, 255), im.getpixel((0, 9)))
+            self.assertEqual((0, 0, 255, 255), im.getpixel((9, 9)))
+            self.assertEqual((0, 0, 255, 255), im.getpixel((5, 9)))
+            self.assertEqual((0, 0, 255, 255), im.getpixel((5, 5)))
+
+            self.assertEqual((255, 0, 0, 255), im.getpixel((0, 10)))
+            self.assertEqual((255, 0, 0, 255), im.getpixel((9, 10)))
+            self.assertEqual((255, 0, 0, 255), im.getpixel((5, 10)))
+            self.assertEqual((255, 0, 0, 255), im.getpixel((0, 15)))
+
+            self.assertEqual((255, 0, 0, 255), im.getpixel((0, 19)))
+            self.assertEqual((255, 0, 0, 255), im.getpixel((9, 19)))
+            self.assertEqual((255, 0, 0, 255), im.getpixel((5, 19)))
+            self.assertEqual((255, 0, 0, 255), im.getpixel((9, 15)))
```

Comment:
Instead of checking all pixels for the border I think we can get away with just checking the four corners (here and in the other tests)

Reply:
You mean checking the corner pixels of the box? 

![4_corners](https://user-images.githubusercontent.com/1443354/152322015-cee09fb1-5723-4d26-a195-58d00c694d71.png)

I recall that initially I had started with this approach, just checking the four corner pixels, but then during the implementation when I had made a mistake, and the rectange got wider, and lthe right side of the box where rendered ourside of the image, the test still had passed, as the top and bottom lines where still covering all four corner pixels. 

![4_corners_no_right_line](https://user-images.githubusercontent.com/1443354/152322067-2e96e67c-c161-4782-8a6c-3550781f7640.png)

This is why I had decided to check the pixels in the middle of the lines as well, as this approach would catch that mistake.

![8_corners_and_middle](https://user-images.githubusercontent.com/1443354/152322106-59ee8c7e-4bda-4328-810b-9b0092b339d1.png)

![8_corners_and_middleno_right_line](https://user-images.githubusercontent.com/1443354/152322121-9f736967-882b-491f-ac65-7c5ee4faeef9.png)

I would suggest to keep these checks to make sure this mistake is covered.

Reply:
Sure, makes sense. We just published a new version that partially addresses the issues this PR fixes, do you mind rebasing onto that? It seems like `ImageChops.difference` in general is not very suitable for ARGB PNGs, unfortunately, but I'm not sure off the top of my head on what would be an easy / better alternative within Pillow.